src/app-layer-smb.c

```diff
@@ -230,95 +38,11 @@ static SuricataFileContext sfc = { &sbcfg };
 static void SMBParserRegisterTests(void);
 #endif
 
-static uint32_t stream_depth = SMB_CONFIG_DEFAULT_STREAM_DEPTH;
-
 void RegisterSMBParsers(void)
 {
-    const char *proto_name = "smb";
-
-    /** SMB */
-    if (AppLayerProtoDetectConfProtoDetectionEnabled("tcp", proto_name)) {
-        AppLayerProtoDetectRegisterProtocol(ALPROTO_SMB, proto_name);
-        if (SMBRegisterPatternsForProtocolDetection() < 0)
-            return;
-
-        rs_smb_init(&sfc);
-
-        if (RunmodeIsUnittests()) {
```

Comment:
If there are no C tests is this conditional still required? If SMB is used by other protos, is that when we need this?
This has not been translated to Rust

Reply:
I don't see a reason to keep it if there are no tests.